### ISSUE TITLE
File filters not working on MacOS

### ISSUE THREAD

**Issue by SolarLune** [gen2brain/dlgs#14]
Hello!

I've noticed a couple of issues with MacOS's File() (and by extension FileMulti()) functions' file filtering. 

- The filter argument only works for a single file extension - you can't pass, say, "jpg png" to filter out any non-image files.

- The extension filter argument MacOS expects (`jpg`, for example) isn't the same format as what Linux and Windows takes (`*.jpg`).

I've submitted a PR to resolve these issues - #13. I also expounded on the documentation to give an idea of what a cross-platform filter argument would look like. I just wanted to create an issue here to explain it a bit in case you had some questions or wanted to discuss it a bit.

**Comment by gen2brain** [comment]
Yes, filters are messy, not just between OSes but also between Qt and GTK (zenity and qarma). My idea was to follow just one style and to handle internally all cases, but never got to do it.

**Comment by JunLog** [comment]
# File filters is doesn't working on Windows10,
go version go1.16.4 windows/amd64
Microsoft Windows [version 10.0.19041.450]

```go
path, is_ok, err := dlgs.File("Select file", "*.png", false)
    if err != nil {
        panic(err)
    }
    fmt.Println(path, is_ok)
```
